packages/harmony/src/components/text/Text.tsx

```diff
@@ -72,10 +75,20 @@ export const Text = forwardRef(
 
     const Tag: ElementType = asChild ? Slot : tag ?? variantTag ?? 'span'
 
-    return (
+    const textElement = (
       <Tag ref={ref} css={css} {...other}>
         {children}
       </Tag>
     )
+
+    if (contextVariant && !propVariant) {
+      return textElement
+    }
+
+    return (
+      <TextContext.Provider value={{ variant }}>
```

Comment:
Interesting choice not to extend the context to each of the props. I could see wanting the ability to set an inherited context with any combination of props. Do you think that's an anti-pattern?

Currently, it would behave this way, right?
```
<Text variant='title' strength='strong'>
	strong title
	<Text size='s'>
		regular strength small title
	</Text>
</Text>
```

I might expect the inner text to be strong as well given the parent props.

Reply:
I believe it will have strong text because of css cascade. But it is a bit confusing that we have css cascade vs context cascade

Reply:
If it works as expected, I'm fine either way, though I do feel it might be easier to use and reason about if the behavior is defined explicitly here